app/grandchallenge/components/tasks.py

```diff
@@ -979,3 +981,40 @@ def civ_value_to_file(*, civ_pk):
         )
         civ.value = None
         civ.save()
+
+
+def _get_sitk_image(image_file):
+    with NamedTemporaryFile() as tmp:
+        for chunk in image_file.file.chunks():
+            tmp.write(chunk)
+        tmp.flush()
+        tmp.seek(0)
+        loaded_image = load_sitk_image(Path(tmp.name))
+        simple_itk_image = SimpleITKImage(
+            image=loaded_image,
+            name=tmp.name,
+            consumed_files=set(),
+            spacing_valid=True,
+        )
+        return simple_itk_image
+
+
+@shared_task(**settings.CELERY_TASK_DECORATOR_KWARGS["acks-late-2xlarge"])
+def validate_voxel_values(*, civ_pk):
+    civ = get_model_instance(
+        pk=civ_pk,
+        app_label="components",
+        model_name="componentinterfacevalue",
+    )
+
+    if civ.image.segments is None:
+        for file in civ.image.files.all():
+            if not file.image_type == ImageFile.IMAGE_TYPE_MHD:
+                continue
+            simple_itk_image = _get_sitk_image(file)
```

Comment:
This still won't work for mhd/(z)raw pairs. You can enable this mode in the grand challenge settings (I think) to try it out. You need to create a temporary directory in this method, then download the files, then load the mhd.

Reply:
The only thing I seem to able to find is that we use the mhd image builder from panimg, which always saves the output as mha. I can't find any setting that changes this, though I do see an admin filter for zraw/mhd.

Reply:
Hmm, yeah, I guess we removed it when migrating to panimg. If you can monkey patch panimg and change the extension here from `.mha` to `mhd` it should do it https://github.com/DIAGNijmegen/rse-panimg/blob/9e46f1f5c4bcf892b26f58fddc140808a86bdcd8/panimg/models.py#L367